site/docs/hosting/vercel.md

```diff
@@ -0,0 +1,117 @@
+# Hosting: Vercel Serverless Functions
+
+This tutorial will guide you on how to deploy your bot to [Vercel](https://vercel.com/) by using [Vercel Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions), assuming that you already have a [Vercel](https://vercel.com) account.
+
+## Project Structure
+
+The only prerequisite to get started with **Vercel Serverless Functions** is to move your code to the `api/` directory as shown below.
+You can also see [Vercel's documentation](https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions) for more on this.
+
+```asciiart:no-line-numbers
+.
+├── node_modules/
+├── build/
+├── api/
+│   └── bot.ts
+├── package.json
+├── package-lock.json
+└── tsconfig.json
+```
+
+If you are using TypeScript, you might as well want to install `@vercel/node` as a dev dependency, but it is not mandatory to follow this guide.
+
+## Configuring Vercel
+
+The next step is to create a `vercel.json` file at the top level of your project.
+For our example structure, its content would be:
+
+```json
+{
+  "functions": {
+    "api/bot.ts": {
+      "memory": 1024,
+      "maxDuration": 10
+    }
+  }
+}
+```
+
+> Vercel's free subscription has restrictions on quotas, which we need to enable in the configuration file, `vercel.json` ([grammY Examples](https://github.com/grammyjs/examples/blob/main/vercel-bot/api/index.ts)).
```

Comment:
This is unclear. Vercel's free subscription has restrictions on quotas which we need to enable what?

Reply:
Yes, that's the point: this [this vercel example] (https://github.com/grammyjs/examples/blob/main/vercel-bot/api/index.ts) says that and to avoid possible errors I used the same config in my project. To be honest, I can't verify for myself what would happen if you don't apply the `memory` and `maxDuration` constraints in the `vercel.json` file.

In any case, to avoid any possible problem when using Vercel's free plan, I have left that comment the same as in the example that I am telling you.

Reply:
Oh you could then rephrase it, maybe like this:

```suggestion
> If you want to use Vercel's free subscription, your `memory` and `maxDuration` might look like above because of its limits.
```

Reply:
Sure, good idea!